pkg/mutation/system.go

```diff
@@ -244,3 +245,14 @@ func mutateErr(err error, uid uuid.UUID, mID types.ID, obj *unstructured.Unstruc
 		obj.GetNamespace(),
 		obj.GetName())
 }
+
+func mutatorMatchesSource(mut types.Mutator, source types.SourceType) bool {
```

Comment:
This is probably better to be defined as part of the match criteria API object, since that way we get it for free for validation, mutation, and all callers of those two functions